spec/features/adventure_spec.rb

```diff
@@ -1,30 +1,180 @@
+
+  context "when logging in/creating an account" do
+    it "does not allow you to log in with incorrect credentials" do
+      visit user_session_path
+
+      fill_in "Email", with: user.email
+      fill_in "Password", with: "not password"
+      click_on "Log in"
+
+      expect(page).to have_content("Invalid Email or password.")
+    end
+
+    it "allows you to log in with the correct credentials" do
+      visit user_session_path
+      fill_in "Email", with: user.email
+      fill_in "Password", with: user.password
+      click_on "Log in"
+
+      expect(page).to have_content("Signed in successfully")
+      expect(page).to have_current_path(my_adventures_path)
+    end
+
+    it "rejects a new account with mismatched passwords" do
+      visit new_user_registration_path
+      fill_in "user_email", with: "email@example.com"
+      fill_in "user_password", with: "abc123"
+      fill_in "user_password_confirmation", with: "not the same"
+      click_on "Sign up"
+
+      expect(page).to have_content("Password confirmation doesn't match Password")
+    end
+
+    it "allows you to create an account" do
+      visit new_user_registration_path
+      fill_in "user_email", with: "email@example.com"
+      fill_in "user_password", with: "abc123"
+      fill_in "user_password_confirmation", with: "abc123"
+      click_on "Sign up"
+
+      expect(page).to have_content("You have signed up successfully")
+    end
+  end
+
+  context "when updating an account's details" do
+    before do
+      login_as(user, :scope => :user)
+    end
+
+    it "does not let you change your password if you don't know your current password" do
+      visit edit_user_registration_path
+      fill_in "user_password", with: "newpass"
+      fill_in "user_password_confirmation", with: "newpass"
+      fill_in "user_current_password", with: "notthepassword"
+      click_on "Update"
+
+      expect(page).to have_content("Current password is invalid")
+    end
+
+    it "does not let you change your password if password confirmation doesn't match" do
+      visit edit_user_registration_path
+      fill_in "user_password", with: "newpass"
+      fill_in "user_password_confirmation", with: "newpasswrong"
+      fill_in "user_current_password", with: "password"
+      click_on "Update"
+
+      expect(page).to have_content("Password confirmation doesn't match Password")
+    end
+
+    it "allows you to change your password if everything is correct" do
+      visit edit_user_registration_path
+      fill_in "user_password", with: "newpass"
+      fill_in "user_password_confirmation", with: "newpass"
+      fill_in "user_current_password", with: "password"
+      click_on "Update"
+
+      expect(page).to have_content("Your account has been updated successfully.")
+      expect(page).to have_current_path("/")
+    end
+
+  end
+
   context "when not logged in" do
+    before do
+      adventure_password.update_attributes(has_password: true, password: "password")
+    end
```

Comment:
possible to move the `adventure_password` definition to this context block and avoid the need for `update_attributes` here?